.github/workflows/documentation.yml

```diff
@@ -0,0 +1,146 @@
+---
+name: Documentation
+
+# Builds Markdown documentation for core and addons.
+
+on:
+  push:
+    branches:
+      - main
+      - releasebranch_*
+  pull_request:
+
+permissions: {}
+
+jobs:
+  ubuntu:
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+      cancel-in-progress: true
+
+    runs-on: ubuntu-22.04
+    env:
+      PYTHONWARNINGS: always
+
+    steps:
+      - name: Checkout core
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: OSGeo/grass
+          path: grass
+
+      - name: Get dependencies
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install -y wget git gawk findutils
+          xargs -a <(awk '! /^ *(#|$)/' "grass/.github/workflows/apt.txt") -r -- \
+              sudo apt-get install -y --no-install-recommends --no-install-suggests
+
+      - name: Set up Python
+        uses: actions/setup-python@42375524e23c412d93fb67b49958b491fce71c38 # v5.4.0
+        with:
+          python-version: "3.12"
+          cache: pip
+
+      - name: Create installation directory
+        run: |
+          mkdir "$HOME/install"
+
+      - name: Set number of cores for compilation
+        run: |
+          echo "MAKEFLAGS=-j$(nproc)" >> "$GITHUB_ENV"
+
+      - name: Set LD_LIBRARY_PATH for compilation
+        run: |
+          echo "LD_LIBRARY_PATH=$HOME/install/lib" >> "$GITHUB_ENV"
+
+      - name: Build core
+        run: |
+          cd grass
+          ../grass/.github/workflows/build_ubuntu-22.04.sh "$HOME/install"
+
+      - name: Add the bin directory to PATH
+        run: |
+          echo "$HOME/install/bin" >> "$GITHUB_PATH"
+
+      - name: Print installed versions
+        if: always()
+        run: ./grass/.github/workflows/print_versions.sh
+
+      - name: Test executing of the grass command
+        run: ./grass/.github/workflows/test_simple.sh
+
+      - name: Set version variables
+        run: |
+          cd grass
+          eval $(./utils/update_version.py status --bash)
+          echo "MAJOR=$MAJOR" >> $GITHUB_ENV
+          echo "MINOR=$MINOR" >> $GITHUB_ENV
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+          echo "YEAR=$YEAR" >> $GITHUB_ENV
+
+      - name: Checkout addons
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: OSGeo/grass-addons
+          ref: grass${{ env.MAJOR }}
+          path: grass-addons
+
+      - name: Compile addons
+        continue-on-error: true
+        run: |
+          ./grass-addons/utils/cronjobs_osgeo_lxd/compile_addons_git.sh \
+            "$MAJOR" \
+            "$MINOR" \
+            $(pwd)/grass-addons/src \
+            $(grass --config path) \
+            $(pwd)/addons-build-dir \
+            grass
```

Comment:
I'm trying to get a rough workflow where the main part works and the addons may work. I want to continue-on-error in this step and others because they are currently failing or at least not delivering what they should.

In the meantime, #5177 and #5175 are fixing the issues I identified so far and that should be enough to remove continue-on-error here. These are now sort of chicken and egg, the other PRs fix the build, but this PR actually builds in CI, possibly showing how the other two work. We can merge them in any order and I can then go back and improve this workflow. This was my original idea before I created the other PRs.